modules/tinymce/src/plugins/lists/main/ts/actions/ToggleList.ts

```diff
@@ -173,7 +173,12 @@ const getSelectedTextBlocks = (editor: Editor, rng: Range, root: Node): HTMLElem
 
     const nextSibling = node.nextSibling;
     if (BookmarkManager.isBookmarkNode(node)) {
-      if (NodeType.isListNode(nextSibling) || NodeType.isTextBlock(editor, nextSibling) || (!nextSibling && node.parentNode === root)) {
+      if (!nextSibling && node.parentNode === root) {
+        node.previousSibling?.appendChild(node);
```

Comment:
This looks a bit strange. The previous sibling could be anything. What happens if it's a table?

Reply:
@spocke , I changed the logic, have a check